docs/HowToFAQ/FAQMultClauses.md

```diff
@@ -0,0 +1,18 @@
+---
+title: When a lemma has multiple ensures clauses, I’m finding that they interact, when I expected them to be independent.  For example, commenting out one of them can make another one not verify.  How should I think about this?
+---
+
+## Question
+
+When a lemma has multiple ensures clauses, I’m finding that they interact, when I expected them to be independent.  For example, commenting out one of them can make another one not verify.  How should I think about this?
+
+## Answer
+
+Multiple ensures clauses, such as `ensures P ensures Q ensures R` is equivalent to the conjunction, in order, of the ensures predicates: `ensures P && Q && R`.
+The order is important if an earlier predicate is needed to be sure that a later one is well defined. For example the two predicates in `ensures a != null ensures a.Length > 0`
+must be in that order because `a.Length` is well-defined only if `a != null`.
+
+In addition, sometimes one ensures clause serves as a lemma for the next one. Without the earlier clause, Dafny can have trouble proving the second one, even though it is valid.
```

Comment:
Lemma has a specific meaning in this context, so this might cause confusion?

Reply:
Reworded.